### PR TITLE
Add missing interface filters

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

Add implementation for interface relationship filter as `actedInConnection_SOME` / `actors_SOME` when the source type is an Interface.

## Complexity

Complexity: Medium

Closes 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

File: packages/graphql/src/translate/queryAST/factory/FilterFactory.ts [hunk 1]
```diff
@@ -258,106 +288,115 @@ export class FilterFactory {
...
+        return this.wrapMultipleFiltersInLogical(filters);
+    }
 
-                    return this.createRelationshipFilter(relationship, value as GraphQLWhereArg, {
-                        isNot,
-                        operator,
+    public createNodeFilters(entity: EntityAdapter, whereFields: Record<string, any>): Filter[] {
+        if (isInterfaceEntity(entity)) {
+            throw new Error("Cannot query interface entity directly");
+            //return this.createInterfaceNodeFilters(entity, whereFields);
```
Comment (mjfwebb) [inline packages/graphql/src/translate/queryAST/factory/FilterFactory.ts]:
```suggestion
            throw new Error("Cannot query interface entity directly");
```

File: packages/graphql/src/translate/queryAST/factory/FilterFactory.ts [hunk 2]
```diff
@@ -524,4 +638,28 @@ export class FilterFactory {
...
+                if (isAggregate || isConnection) {
+                    return true;
+                }
+                const relationshipDeclaration = entity.findRelationshipDeclarations(fieldName);
+                if (relationshipDeclaration) {
+                    return true;
+                }
+                return false;
+            });
+            return !doesContainsFieldsNotOptimizable;
```
Comment (mjfwebb) [inline packages/graphql/src/translate/queryAST/factory/FilterFactory.ts]:
```suggestion
            const containsUnoptimizableFields = Object.keys(where.node).some((field) => {
                const { fieldName, isAggregate, isConnection } = parseWhereField(field);
                if (isAggregate || isConnection) {
                    return true;
                }
                const relationshipDeclaration = entity.findRelationshipDeclarations(fieldName);
                if (relationshipDeclaration) {
                    return true;
                }
                return false;
            });
            return !containsUnoptimizableFields;
```

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟩 query.QueryWithNestedIn | 13254 | 14922 | 55 | 66 | 1180 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 37 | 45 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 47 | 65 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 60 | 75 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 40 | 63 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2140 | 2244 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2227 | 2238 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 36 | 39 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 31 | 37 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2934 | 2488 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 43 | 60 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 78 | 110 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 139 | 163 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 57 | 63 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 139 | 184 | 3003 |
|  connections.Connection | 14049 | 14049 | 62 | 128 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 160 | 168 | 4516 |
|  create.SimpleMutation | 7 | 7 | 49 | 74 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 24 | 23 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 185 | 210 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 173 | 219 | 2040 |
|  2871.NestedRelationshipFilter | 23146 | 23076 | 55 | 58 | 4455 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 34 | 45 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 69 | 85 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 36 | 38 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 48 | 56 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 18 | 25 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 48 | 45 | 2174 |
|  query.QueryWhere | 9699 | 9688 | 30 | 42 | 2170 |
|  query.SimpleQueryWithNestedWhere | 9866 | 9855 | 44 | 54 | 2170 |
|  query.Nested | 10095502 | 10095502 | 6577 | 7284 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6668 | 6928 | 2004000 |
|  query.OrFilterOnRelationships | 42033 | 42239 | 152 | 172 | 1629 |
|  query.OrFilterOnRelationshipsAndNested | 35720 | 36360 | 189 | 219 | 1629 |
| 🟩 query.QueryWithNestedIn | 13254 | 14922 | 55 | 66 | 1180 |
|  query.NestedConnectionWhere | 9801 | 9801 | 50 | 69 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10006 | 9843 | 68 | 87 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18914 | 18752 | 115 | 166 | 1640 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 146 | 186 | 1134 |
|  query.Fulltext | 64 | 64 | 28 | 36 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 40 | 43 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 46 | 56 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 64 | 72 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14175 | 14179 | 57 | 63 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14175 | 14179 | 97 | 110 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14559 | 14756 | 98 | 116 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14485 | 14537 | 107 | 155 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 79 | 110 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14452 | 14477 | 114 | 126 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 154 | 60 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 98 | 82 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 104 | 82 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 64 | 65 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 89 | 68 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 299 | 295 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 232 | 279 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 98 | 108 | 2002 |

</details>
        
Old Schema Generation: 26.193s
Schema Generation: 25.538s
Old Subgraph Schema Generation: 28.415s
Subgraph Schema Generation: 29.013s

**Review by mjfwebb — CHANGES_REQUESTED** [review]

**Review by mjfwebb — APPROVED** [review]

**Pull request merged**